File: client/src/api/sales.ts
import { api } from './axios';
import type { Sale, ListSalesParams, CreateSaleInput } from '@/types/sale';
import type { ApiPaginated } from '@/types/api';

export const saleApi = {
  list: (params: ListSalesParams = {}) =>
    api
      .get<ApiPaginated<Sale>>('/client/sales', { params })
      .then((r) => r.data),

  get: (id: string) =>
    api.get<{ data: Sale }>(`/client/sales/${id}`).then((r) => r.data.data),

  create: (payload: CreateSaleInput) =>
    api.post<{ data: Sale }>('/client/sales', payload).then((r) => r.data.data),

  void: (id: string, reason: string) =>
    api
      .post<{ data: Sale }>(`/client/sales/${id}/void`, { reason })
      .then((r) => r.data.data),

  reprint: (id: string) =>
    api
      .post<{ data: Sale }>(`/client/sales/${id}/reprint`)
      .then((r) => r.data.data),
};

File: Android/client/lib/api/sales.dart
import 'api_client.dart';

class SalesApi {
  static final _api = BizOsApiClient.instance;

  static Future<List<Map<String, dynamic>>> list({
    Map<String, dynamic>? query,
  }) async => _api.list(await _api.get('/client/sales', query: query));
  static Future<Response<dynamic>> create(Map<String, dynamic> data) =>
      _api.post('/client/sales', data: data);
  static Future<Response<dynamic>> get(String id) =>
      _api.get('/client/sales/$id');
  static Future<Response<dynamic>> voidSale(
    String id, {
    Map<String, dynamic>? data,
  }) => _api.post('/client/sales/$id/void', data: data);
  static Future<Response<dynamic>> reprint(String id) =>
      _api.post('/client/sales/$id/reprint');
}
